### PR TITLE
Moved the path building out of execution time

### PR DESCRIPTION
**Pull request by AndrewBurian**
Substantially less memory allocated per execution

### CODE REVIEW

**Comment by AndrewBurian** [comment]
All the savings.
```
BenchmarkSingleRoute-2      	 3000000	       533 ns/op	     208 B/op	       4 allocs/op
BenchmarkShallowAndWide-2   	 2000000	       876 ns/op	     222 B/op	       6 allocs/op
BenchmarkNarrowAndDeep-2    	  100000	     15729 ns/op	    4432 B/op	       6 allocs/op
BenchmarkFan-2              	 2000000	       848 ns/op	     214 B/op	       6 allocs/op
PASS
ok  	github.com/AndrewBurian/powermux	9.418s
```
to
```
BenchmarkSingleRoute-2      	 3000000	       386 ns/op	     160 B/op	       3 allocs/op
BenchmarkShallowAndWide-2   	 2000000	       685 ns/op	     160 B/op	       3 allocs/op
BenchmarkNarrowAndDeep-2    	  100000	     11378 ns/op	    1792 B/op	       3 allocs/op
BenchmarkFan-2              	 2000000	       618 ns/op	     160 B/op	       3 allocs/op
PASS
ok  	github.com/AndrewBurian/powermux	7.300s
```

File: route.go [hunk 1]
```diff
@@ -261,6 +261,7 @@ func (r *Route) create(path []string) *Route {
 
 	// set the pattern name
 	newRoute.pattern = path[1]
+	newRoute.fullPath = r.fullPath + "/" + path[1]
```
Comment (chrisaxiom) [inline route.go:264]: I just benched this:

```
var strs = []string{"a long string", "another string"}

func BenchmarkStringConcat(b *testing.B) {

	for i := 0; i < b.N; i++ {
		var res string
		for _, s := range strs {
			res = res + "/" + s
		}

		if len(res) > 0 {

		}
	}
}

func BenchmarkJoin(b *testing.B) {
	for i := 0; i < b.N; i++ {
		res := strings.Join(strs, "/")
		if len(res) > 0 {

		}
	}
}

```

Results:

```
BenchmarkStringConcat-4         10000000               126 ns/op              48 B/op          2 allocs/op
BenchmarkJoin-4                 20000000                94.7 ns/op            64 B/op          2 allocs/op
```

Comment (chrisaxiom) [reply route.go:264]: Actually, invalid benchmark, the first one is doing an extra join.  Looks like the manual concat is still better then anything else??? Not sure why.

Comment (chrisaxiom) [reply route.go:264]: https://gist.github.com/dtjm/c6ebc86abe7515c988ec

My results:

```
BenchmarkJoin-4                 20000000               115 ns/op              32 B/op          2 allocs/op
BenchmarkSprintf-4               3000000               552 ns/op              96 B/op          6 allocs/op
BenchmarkConcat-4               20000000                77.0 ns/op             0 B/op          0 allocs/op
```

**Review by chrisaxiom — APPROVED** [review]

**Pull request merged**